src/utils.py

```diff
@@ -62,23 +63,26 @@ def generate_secret(length: int = 32) -> str:
 
     Returns:
         An alphanumeric string of the specified length.
+
+    Raises:
+        ValueError: If requested length is shorter than 4.
     """
     if length < 4:
-        raise ValueError("Specific secret length '%d' is too short.", length)
+        raise ValueError(f"Specific secret length '{length}' is too short.")
```

Comment:
Question: any reason you changed this back to an f-string? I though this was not the recommended approach?

Reply:
Linter complained about lazy strings. Apparently it is lazy strings for logs and f-strings for exceptions.

Reply:
Good to know.